src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ProvideHttpClientHandlerMaxResponseHeaderLengthValueCorrectlyTests.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Usage.ProvideHttpClientHandlerMaxResponseHeaderLengthValueCorrectly,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Usage.ProvideHttpClientHandlerMaxResponseHeaderLengthValueCorrectly,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+namespace Microsoft.NetCore.Analyzers.Usage.UnitTests
+{
+    public class ProvideHttpClientHandlerMaxResponseHeaderLengthValueCorrectlyTests
+    {
+        [Fact]
+        public async Task CA2262_ProvideCorrectValueFor_HttpClientHandlerMaxResponseHeader_DiagnosticAsync()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                using System.Net.Http;
+
+                public class TestClass {
+                    
+                    static int GetValue() => 1414;
+                    const int val = 121212 * 2;
+
+                    public void TestMethod() {
+
+                        HttpClientHandler handler = new HttpClientHandler()
+                        {
+                            MaxResponseHeadersLength = GetValue()
+                        };
+
+                        HttpClientHandler handler2 = new HttpClientHandler()
+                        {
+                            {|#0:MaxResponseHeadersLength = val|}
+                        };
+
+                        HttpClientHandler handler3 = new HttpClientHandler()
+                        {
+                            {|#1:MaxResponseHeadersLength = 1414|}
+                        };
+
+                        HttpClientHandler handler4 = new HttpClientHandler()
+                        {
+                            MaxResponseHeadersLength  = int.MaxValue
+                        };
+      
+                        SocketsHttpHandler handler5 = new SocketsHttpHandler() 
+                        {
+                            MaxResponseHeadersLength  = int.MaxValue
```

Comment:
I'm guessing this one was supposed to be testing `= val`?

Reply:
Ah I see, the one above is `HttpClientHandler` not `SocketsHttpHandler`, thought it was a duplicate at first